### PR TITLE
Implement String.chars()

### PR DESCRIPTION
**Pull request by JFronny**
Simple implementation inspired by toCharArray.
Seems to work fine from my testing.

### CODE REVIEW

**Comment by JFronny** [comment]
What copyright owner should be specified in the license header of the stream class?

**Comment by konsoletyper** [comment]
> What copyright owner should be specified in the license header of the stream class?

You can specify your own name. Or you can also implement stream with inner class.

**Comment by JFronny** [comment]
> This PR lacks tests

I have implemented a simple test. How would I go about running it? The documentation mentions setup steps, but not how I would actually run the tests.

**Comment by konsoletyper** [comment]
You should follow instructions in https://teavm.org/docs/contributing/testing.html and then run tests using your favourite way. You can just launch `./gradlew test` from console. Personally, I use IDEA for development and just press launch button across test method.

**Comment by JFronny** [comment]
The issue seems to be that the test runner expects Google Chrome to be installed with a specific name and crashes when it can't find it. Using `-Pteavm.tests.browser=browser` and manually opening the browser works, though.

File: classlib/src/main/java/org/teavm/classlib/java/util/stream/intimpl/TStringCharsStream.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+    public TStringCharsStream(TString string) {
+        this.string = Objects.requireNonNull(string);
+    }
+
+    @Override
+    public boolean next(IntPredicate consumer) {
+        while (index < string.length()) {
+            if (!consumer.test(string.charAt(index++))) {
+                break;
```
[the remark this answers is not in the document]
Comment (JFronny) [reply classlib/src/main/java/org/teavm/classlib/java/util/stream/intimpl/TStringCharsStream.java:34]: An equivalent alternative that avoids the unnecessary calculation but still checks after incrementing the index would be
```java
        while (index < string.length()) {
            if (!consumer.test(string.charAt(index++))) {
                return index < string.length();
            }
        }
        return false;
```
but I doubt that is preferred

File: classlib/src/main/java/org/teavm/classlib/java/util/stream/intimpl/TStringCharsStream.java [hunk 2]
```diff
@@ -21,10 +21,11 @@
 
 public class TStringCharsStream extends TSimpleIntStreamImpl {
     private final TString string;
-    private int index = 0;
+    private int index;
 
     public TStringCharsStream(TString string) {
         this.string = Objects.requireNonNull(string);
+        this.index = 0;
```
Comment (konsoletyper) [inline classlib/src/main/java/org/teavm/classlib/java/util/stream/intimpl/TStringCharsStream.java:28]: You don't need this. Java spec guarantees to initialize int fields with 0

Comment (JFronny) [reply classlib/src/main/java/org/teavm/classlib/java/util/stream/intimpl/TStringCharsStream.java:28]: I know, but explicitly stating that we start at index 0 seemed nicer.

**Pull request merged**